src/protocol/mod.rs

```diff
@@ -449,9 +457,14 @@ impl WebSocketContext {
                     } else {
```

Comment:
I think the condition below could be use in the `else if` block.

Reply:
Can we still make it part of this `if`? :)
I.e. something like:

```
} else if !self.config.accept_unmasked_frames {
    return Err(....
}
```

Reply:
just to confirm before I push it. Something like below?

```
                    if frame.is_masked() {
                        // A server MUST remove masking for data frames received from a client
                        // as described in Section 5.3. (RFC 6455)
                        frame.apply_mask()
                    } else if !self.config.accept_unmasked_frames {
                        // The server MUST close the connection upon receiving a
                        // frame that is not masked. (RFC 6455)
                        // The only exception here is if the user explicitly accepts given
                        // stream by setting WebSocketConfig.accept_unmasked_frames to true
                        return Err(Error::Protocol(
                            "Received an unmasked frame from client".into(),
                        ));
                    }
```

Reply:
Yes, looks correct.

Reply:
done. now it is perfect:)